### PR TITLE
corrected bug in utf.py and new tests

### PR DESCRIPTION
**Pull request by fonkap**

### CODE REVIEW

File: jawa/util/utf.py [hunk 1]
```diff
@@ -63,18 +65,18 @@ def encode_modified_utf8(u):
...
-            final_string.extend(
+            final_string.extend([
                 (0xE0 | (0x0F & (c >> 12))),
                 (0x80 | (0x3F & (c >> 6))),
-                (0x80 | (0x3F & c))
+                (0x80 | (0x3F & c))]
             )
 
-    return str(final_string)
+    return final_string
```
Comment (TkTech) [inline jawa/util/utf.py]: A cast is required here, we do not want to be returning a `bytearray`

Comment (fonkap) [inline jawa/util/utf.py]: Well, I gently disagree, the logical result of decoding a mutf string in my opinion is a bytearray. Returning a string can be confusing.
But, of course, this is absolutely up to you.

Comment (TkTech) [inline jawa/util/utf.py]: Thinking over this more I'm fine with this change.

File: tests/test_modified_utf8.py [hunk 2]
```diff
@@ -20,3 +22,24 @@ def test_decode_modified_utf8():
...
+    str1 = u'1\x002'
+    bb = encode_modified_utf8(str1)
+    assert '31c08032' == binascii.hexlify(bb), binascii.hexlify(bb)
+
+    # Unicode supplementary character U+10400
+    str1 = u'\uD801\uDC00'
+    bb = encode_modified_utf8(str1)
+    assert 'eda081edb080' == binascii.hexlify(bb), binascii.hexlify(bb)
+
+def test_decode_utf8_1():
```
Comment (TkTech) [inline tests/test_modified_utf8.py]: Would you be able to expand the docstrings on these tests? Happy to merge after that.

**Comment by fonkap** [comment]
Added commit with docs in test methods.
I hope everything is fine now :)

**Pull request merged**